src/main/java/com/google/graphgeckos/dashboard/storage/BuildInfo.java

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.storage;
+
+import com.google.cloud.Timestamp;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Entity;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Unindexed;
+import org.springframework.data.annotation.Id;
+
+/**
+ * This class encapsulates all the information gathered about a specific revision.
+ * Information included: 
+ *    * commit hash
+ *    * timestamp
+ *    * branch
+ *    * information from builders:
+ *          - builder name
+ *          - compile logs
+ *          - compilation status
+ *
+ * This class is used to facilitate Datastore I/O operations through Spring Cloud API,
+ * and to transfer the data from the DataRepository to the REST API component.
+ */
+@Entity(name = "revision")
+public class BuildInfo {
+
+  @Id
+  @Field(name = "commitHash")
+  private final String commitHash;
+
+  @Field(name = "timestamp")
+  private final Timestamp timestamp;
+
+  @Field(name = "branch")
+  private final String branch;
+
+  @Field(name = "builders")
```

Comment:
Jackson (e.g ```@JsonProperty```) annotations are required for parsing jsons automatically. I would suggest to convert my POJO to yours before sending it to cloud storage.